src/event.rs

```diff
@@ -0,0 +1,530 @@
+use crate::{
+	hex_utils, ChannelManager, Error, FilesystemPersister, LdkLiteChainAccess, LdkLiteConfig,
+	NetworkGraph, PaymentInfo, PaymentInfoStorage, PaymentStatus,
+};
+
+#[allow(unused_imports)]
+use crate::logger::{
+	log_error, log_given_level, log_info, log_internal, log_trace, log_warn, FilesystemLogger,
+	Logger,
+};
+
+use lightning::chain::chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator};
+use lightning::chain::keysinterface::KeysManager;
+use lightning::ln::PaymentHash;
+use lightning::routing::gossip::NodeId;
+use lightning::util::events as ldk_events;
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::{MaybeReadable, Readable, ReadableArgs, Writeable, Writer};
+
+use bitcoin::secp256k1::Secp256k1;
+use rand::{thread_rng, Rng};
+use std::collections::{hash_map, VecDeque};
+use std::sync::{Arc, Condvar, Mutex};
+use std::thread;
+use std::time::Duration;
+
+/// The event queue will be persisted under this key.
+pub(crate) const EVENTS_PERSISTENCE_KEY: &str = "events";
+
+/// An event emitted by [`LdkLite`] that should be handled by the user. This will more often than
+/// not wrap a LDK event accessible via the `inner` field.
+#[derive(Debug, Clone)]
+pub enum LdkLiteEvent {
```

Comment:
Duly noted. I introduced the `LdkLite` prefix to all of LDKLite's objects to clearly distinguish them from upstream LDKs variants. This is not so much meant to improve readability here, but to make the downstream users' lives easier, since they suddenly will deal with a set of similar but not identical objects. This is however up for discussion, I'm also not hundred percent sure about this. However, would you be fine deferring this discussion to a more general 'API cleanup' step, e.g., as part of #18?